kolibri/core/errorreports/middleware.py

```diff
@@ -1,12 +1,35 @@
 import logging
 import traceback
+from sys import version
 
+import pkg_resources
 from django.db import IntegrityError
 
 from .constants import BACKEND
 from .models import ErrorReports
 
 
+def get_request_info(request):
+    return {
+        "url": request.build_absolute_uri(),
+        "method": request.method,
+        "headers": dict(request.headers),
+        "body": request.body.decode("utf-8"),
+    }
+
+
+def get_server_info(request):
+    return {"host": request.get_host(), "port": request.get_port()}
+
+
+def get_packages():
+    return {dist.project_name: dist.version for dist in pkg_resources.working_set}
+
+
+def get_python_version():
+    return version.split()[0]
```

Comment:
I think this would only pick the first version of the tuple. See [here](https://github.com/learningequality/kolibri/blob/24186194c23f2f0dd126e72bdc79ed4b5ddd413f/kolibri/core/device/api.py#L202-L204)